File: Trunk/vpPriV100GrupoMundifios/CondPagRQ/Vendas/FichaConverteVendas/VndIsFichaConverteVendas.cs
using Generico;
using Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs;
using Primavera.Extensibility.Sales.Editors;
using Primavera.Platform.Collections;
using System.Collections;
using VndBE100;

namespace condpagrq
{
    public class vndisfichaconvertevendas : FichaConverteVendas
    {
        public override void DepoisDeConverter(PrimaveraOrderedDictionary colDocumentosGerados, ExtensibilityEventArgs e)
        {
            base.DepoisDeConverter(colDocumentosGerados, e);

            if (Module1.VerificaToken("CondPagRQ") == 1)
            {
                VndBEDocumentoVenda DocVendaRqRm = new VndBEDocumentoVenda();

                foreach (DictionaryEntry itdoc in colDocumentosGerados)
                {
                    BasBE100.BasBETiposGcp.TPDocumentos docs = (BasBE100.BasBETiposGcp.TPDocumentos)itdoc.Value;


                    DocVendaRqRm = BSO.Vendas.Documentos.Edita(docs.Filail,docs.Tipodoc,docs.Serie,docs.NumDoc);

                    if (DocVendaRqRm.CondPag + "" != "")
                    {
                        if ((bool)BSO.Base.CondsPagamento.Edita(DocVendaRqRm.CondPag).CamposUtil["CDU_RQ"].Valor == true | (bool)BSO.Base.CondsPagamento.Edita(DocVendaRqRm.CondPag).CamposUtil["CDU_RM"].Valor == true)
                        {
                            DocVendaRqRm.DataVenc = Module1.NovaDataVencimento(DocVendaRqRm.DataDoc, DocVendaRqRm.CondPag, DocVendaRqRm.TipoEntidade, DocVendaRqRm.Entidade);
                            DocVendaRqRm.CamposUtil["CDU_AlteradaDataVenc"].Valor = 1;
                            BSO.Vendas.Documentos.Actualiza(DocVendaRqRm);
                        }
                    }
                }
            }
        }
    }
}

File: Branches/VBInicial/vpPriV100GrupoMundifios/CondPagRQ/Vendas/FichaConverteVendas/VndIsFichaConverteVendas.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports Primavera.Extensibility.Sales.Editors
Imports Primavera.Extensibility.BusinessEntities
Imports Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs
Imports VndBE100
Imports Vimaponto.PrimaveraV100.Clientes.GrupoMundifios.Generico

Namespace CondPagRQ
    Public Class VndIsFichaConverteVendas
        Inherits FichaConverteVendas

        Public Overrides Sub DepoisDeConverter(colDocumentosGerados As Primavera.Platform.Collections.PrimaveraOrderedDictionary, e As ExtensibilityEventArgs)
            MyBase.DepoisDeConverter(colDocumentosGerados, e)

            If Module1.VerificaToken("CondPagRQ") = 1 Then

                Dim DocVendaRQRM As New VndBEDocumentoVenda

                Dim itdoc As Object

                For Each itdoc In colDocumentosGerados

                    DocVendaRQRM = BSO.Vendas.Documentos.Edita(itdoc.Filial, itdoc.Tipodoc, itdoc.Serie, itdoc.NumDoc)

                    If DocVendaRQRM.CondPag & "" <> "" Then

                        If BSO.Base.CondsPagamento.Edita(DocVendaRQRM.CondPag).CamposUtil("CDU_RQ").Valor = True Or BSO.Base.CondsPagamento.Edita(DocVendaRQRM.CondPag).CamposUtil("CDU_RM").Valor = True Then

                            DocVendaRQRM.DataVenc = Module1.NovaDataVencimento(DocVendaRQRM.DataDoc, DocVendaRQRM.CondPag, DocVendaRQRM.TipoEntidade, DocVendaRQRM.Entidade)
                            DocVendaRQRM.CamposUtil("CDU_AlteradaDataVenc").Valor = 1
                            BSO.Vendas.Documentos.Actualiza(DocVendaRQRM)
                        End If

                    End If
                Next
            End If
        End Sub

    End Class
End Namespace